File: day4/cxx/puzzle1.cxx
#include <algorithm>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

#include "lines.hpp"
#include "util.hpp"

using namespace std;

const auto solve()
{
    ifstream f{"../input.txt"};

    auto matches = 0u;

    for (auto & line: lines(f))
    {
        auto ranges = get_range_pair(line);

        if (contains(get<0>(ranges), get<1>(ranges)) || contains(get<1>(ranges), get<0>(ranges)))
        {
            matches++;
        }
    }

    return matches;
}

int main()
{
    auto solution = solve();
    cout << solution << endl;
}

File: day4/py/puzzle1.py
#!/usr/bin/env python

from util import get_range_pair, contains

def solve():
    matches = 0

    with open('../input.txt', 'r') as f:
        for line in f:
            ranges = get_range_pair(line)

            if contains(ranges[0], ranges[1]) or contains(ranges[1], ranges[0]):
                matches += 1

    return matches

if __name__ == "__main__":
    solution = solve()
    print(solution, )

